### ISSUE TITLE
Cannot run on MacOS 14 - zsh: segmentation fault

### ISSUE THREAD

**Issue by nicraMarcin** [saturneric/GpgFrontend#178]
I cannot run on this when I have Language as EN and Region PL

```shell
 $> /Applications/GpgFrontend.app/Contents/MacOS/GpgFrontend -l debug
[20240903 10:45:03.194] [core] [D] [5251427] - a new global singleton storage collection created, address: 0x600002a914b0
[20240903 10:45:03.194] [core] [D] [5251427] - cannot find channel object, channel: 0
[20240903 10:45:03.194] [core] [I] [5251427] - app path:  "/Applications/GpgFrontend.app/Contents/MacOS"
[20240903 10:45:03.194] [core] [I] [5251427] - app working path:  "/Users/mrc"
[20240903 10:45:03.194] [core] [I] [5251427] - app data path:  "/Users/mrc/Library/Application Support/GpgFrontend"
[20240903 10:45:03.194] [core] [I] [5251427] - app log path:  "/Users/mrc/Library/Application Support/GpgFrontend/logs"
[20240903 10:45:03.194] [core] [D] [5251427] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x6000026c8120
[20240903 10:45:03.194] [core] [D] [5251427] - register channel object to instances map, channel: 0, address: 0x6000026c8120
[20240903 10:45:03.194] [core] [D] [5251427] - set channel: 0 success, current channel object address: 0x6000026c8120
[20240903 10:45:03.195] [core] [D] [5251427] - cannot find channel object, channel: 0
[20240903 10:45:03.195] [core] [D] [5251427] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x6000006bc800
[20240903 10:45:03.195] [core] [D] [5251427] - register channel object to instances map, channel: 0, address: 0x6000006bc800
[20240903 10:45:03.195] [core] [D] [5251427] - set channel: 0 success, current channel object address: 0x6000006bc800
[20240903 10:45:03.196] [core] [D] [5251427] - cannot find channel object, channel: 0
[20240903 10:45:03.196] [core] [D] [5251427] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x6000026b6190
[20240903 10:45:03.196] [core] [D] [5251427] - register channel object to instances map, channel: 0, address: 0x6000026b6190
[20240903 10:45:03.196] [core] [D] [5251427] - set channel: 0 success, current channel object address: 0x6000026b6190
[20240903 10:45:03.196] [core] [D] [5251427] - dear module manager, is all module registered? answer:  false
[20240903 10:45:03.198] [core] [D] [5251427] - cannot find channel object, channel: 0
[20240903 10:45:03.198] [core] [D] [5251427] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x6000003abaa0
[20240903 10:45:03.198] [core] [D] [5251427] - register channel object to instances map, channel: 0, address: 0x6000003abaa0
[20240903 10:45:03.198] [core] [D] [5251427] - set channel: 0 success, current channel object address: 0x6000003abaa0
[20240903 10:45:03.203] [core] [D] [5251463] - a new need register module:  "libmod_gpg_info.dylib"
[20240903 10:45:03.203] [core] [D] [5251463] - (+) module:  "com.bktus.gpgfrontend.module.gnupg_info_gathering" registering...
[20240903 10:45:03.203] [module] [D] [5251463] - "gnupg info gathering module registering..."
[20240903 10:45:03.204] [ui] [D] [5251463] - module  "com.bktus.gpgfrontend.module.gnupg_info_gathering" registering translator reader...
[20240903 10:45:03.204] [core] [D] [5251463] - (+) module:  "com.bktus.gpgfrontend.module.gnupg_info_gathering" registered.
[20240903 10:45:03.204] [core] [D] [5251463] - cannot find channel object, channel: 0
[20240903 10:45:03.204] [core] [D] [5251463] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x600001a82fd0
[20240903 10:45:03.204] [core] [D] [5251463] - register channel object to instances map, channel: 0, address: 0x600001a82fd0
[20240903 10:45:03.204] [core] [D] [5251463] - set channel: 0 success, current channel object address: 0x600001a82fd0
[20240903 10:45:03.204] [core] [D] [5251463] - (*) module:  "com.bktus.gpgfrontend.module.gnupg_info_gathering" activating...
[20240903 10:45:03.204] [core] [D] [5251463] - (*) module:  "com.bktus.gpgfrontend.module.gnupg_info_gathering" activated.
[20240903 10:45:03.206] [core] [D] [5251463] - a new need register module:  "libmod_paper_key.dylib"
[20240903 10:45:03.207] [core] [D] [5251463] - (+) module:  "com.bktus.gpgfrontend.module.paper_key" registering...
[20240903 10:45:03.207] [module] [D] [5251463] - "paper key module registering"
[20240903 10:45:03.207] [core] [D] [5251463] - (+) module:  "com.bktus.gpgfrontend.module.paper_key" registered.
[20240903 10:45:03.207] [core] [D] [5251463] - (*) module:  "com.bktus.gpgfrontend.module.paper_key" activating...
[20240903 10:45:03.207] [core] [D] [5251463] - (*) module:  "com.bktus.gpgfrontend.module.paper_key" activated.
[20240903 10:45:03.210] [core] [D] [5251463] - a new need register module:  "libmod_pinentry.dylib"
[20240903 10:45:03.210] [core] [D] [5251463] - (+) module:  "com.bktus.gpgfrontend.module.pinentry" registering...
[20240903 10:45:03.210] [module] [D] [5251463] - "pinentry module registering"
[20240903 10:45:03.210] [core] [D] [5251463] - (+) module:  "com.bktus.gpgfrontend.module.pinentry" registered.
[20240903 10:45:03.210] [core] [D] [5251463] - (*) module:  "com.bktus.gpgfrontend.module.pinentry" activating...
[20240903 10:45:03.211] [core] [D] [5251463] - (*) module:  "com.bktus.gpgfrontend.module.pinentry" activated.
[20240903 10:45:03.213] [core] [D] [5251463] - a new need register module:  "libmod_ver_check.dylib"
[20240903 10:45:03.213] [core] [D] [5251463] - (+) module:  "com.bktus.gpgfrontend.module.version_checking" registering...
[20240903 10:45:03.213] [module] [I] [5251463] - "version checking module registering"
[20240903 10:45:03.213] [ui] [D] [5251463] - module  "com.bktus.gpgfrontend.module.version_checking" registering translator reader...
[20240903 10:45:03.213] [core] [D] [5251463] - (+) module:  "com.bktus.gpgfrontend.module.version_checking" registered.
[20240903 10:45:03.213] [core] [D] [5251463] - (*) module:  "com.bktus.gpgfrontend.module.version_checking" activating...
[20240903 10:45:03.213] [module] [I] [5251463] - "version checking module activating"
[20240903 10:45:03.213] [core] [D] [5251463] - (*) module:  "com.bktus.gpgfrontend.module.version_checking" activated.
[20240903 10:45:03.213] [core] [D] [5251463] - all modules are loaded into memory.
[20240903 10:45:03.217] [core] [I] [5251427] - detected gpgconf path:  "/opt/homebrew/bin/gpgconf"
[20240903 10:45:03.217] [core] [I] [5251427] - detected gnupg path:  "/opt/homebrew/Cellar/gnupg/2.4.5_1/bin/gpg"
[20240903 10:45:03.312] [default] [I] [5251427] - current system default locale:  "en_PL"
[20240903 10:45:03.312] [default] [I] [5251427] - locale settings from config:  ""
[20240903 10:45:03.312] [core] [D] [5251526] - monitor: core env is still initializing, waiting...
[20240903 10:45:03.312] [default] [I] [5251427] - application's target locale:  "en_PL"
[20240903 10:45:03.312] [core] [D] [5251525] - set custom gpgconf path:  "/opt/homebrew/Cellar/gnupg/2.4.5_1/bin/gpg"
[20240903 10:45:03.313] [default] [D] [5251427] - locale info: 
[20240903 10:45:03.314] [ui] [D] [5251427] - ui init is done, but core doesn't, going to wait for core...
[20240903 10:45:03.314] [ui] [D] [5251427] - need to wait for env checking process
[20240903 10:45:03.321] [core] [D] [5251525] - gpg context: offline mode: 0
[20240903 10:45:03.321] [core] [D] [5251525] - gpg context: auto import missing key: 0
[20240903 10:45:03.321] [core] [D] [5251525] - got gpgme version version from rt:  "1.23.2"
[20240903 10:45:03.321] [core] [D] [5251525] - set context database path to "/Users/mrc/.gnupg"
[20240903 10:45:03.321] [core] [D] [5251525] - ctx set engine info, db path:  "/Users/mrc/.gnupg" , app path:  "/opt/homebrew/Cellar/gnupg/2.4.5_1/bin/gpg"
[20240903 10:45:03.329] [core] [D] [5251526] - monitor: core env is still initializing, waiting...
[20240903 10:45:03.331] [core] [D] [5251525] - set custom gpgconf path:  "/opt/homebrew/Cellar/gnupg/2.4.5_1/bin/gpg"
[20240903 10:45:03.342] [core] [D] [5251525] - gpg context: offline mode: 0
[20240903 10:45:03.342] [core] [D] [5251525] - gpg context: auto import missing key: 0
[20240903 10:45:03.342] [core] [D] [5251525] - got gpgme version version from rt:  "1.23.2"
[20240903 10:45:03.342] [core] [D] [5251525] - set context database path to "/Users/mrc/.gnupg"
[20240903 10:45:03.342] [core] [D] [5251525] - ctx set engine info, db path:  "/Users/mrc/.gnupg" , app path:  "/opt/homebrew/Cellar/gnupg/2.4.5_1/bin/gpg"
[20240903 10:45:03.348] [core] [D] [5251526] - monitor: core env is still initializing, waiting...
[20240903 10:45:03.351] [core] [D] [5251525] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x6000026c8e10
[20240903 10:45:03.351] [core] [D] [5251525] - register channel object to instances map, channel: 0, address: 0x6000026c8e10
[20240903 10:45:03.351] [core] [D] [5251525] - set channel: 0 success, current channel object address: 0x6000026c8e10
[20240903 10:45:03.351] [core] [D] [5251525] - cannot find channel object, channel: 0
[20240903 10:45:03.352] [core] [D] [5251525] - set channel object, type: PN11GpgFrontend13ChannelObjectE in channel: 0, address: 0x6000026a3810
[20240903 10:45:03.352] [core] [D] [5251525] - register channel object to instances map, channel: 0, address: 0x6000026a3810
[20240903 10:45:03.352] [core] [D] [5251525] - set channel: 0 success, current channel object address: 0x6000026a3810
[20240903 10:45:03.364] [core] [D] [5251526] - monitor: core env is still initializing, waiting...
[20240903 10:45:03.383] [core] [D] [5251526] - monitor: core env is still initializing, waiting...
[20240903 10:45:03.390] [ui] [D] [5251427] - ui is ready to wait for env initialized, env_state: 0
[20240903 10:45:03.392] [core] [D] [5251525] - basic env checking finished, including gpgme, ctx, and key infos
[20240903 10:45:03.400] [core] [D] [5251526] - module manager report, need register:  4 registered 4
[20240903 10:45:03.400] [core] [D] [5251526] - monitor: good, all module are registered.
[20240903 10:45:03.400] [core] [D] [5251526] - monitor: core is fully initialized, sending signal to ui...
[20240903 10:45:03.439] [ui] [D] [5251427] - ui caught signal: core fully loaded
[20240903 10:45:03.439] [core] [D] [5251427] - monitor task ended, call back to main thead.
[20240903 10:45:03.477] [ui] [D] [5251427] - application fully initialized...
[20240903 10:45:03.477] [default] [W] [5251427] - QString::arg: Argument missing: :/i18n/ModuleGnuPGInfoGathering.en_PL.qm loading, locale: en_PL, not found, com.bktus.gpgfrontend.module.gnupg_info_gathering
[20240903 10:45:03.477] [module] [W] [5251427] - ":/i18n/ModuleGnuPGInfoGathering.en_PL.qm loading, locale: en_PL, not found"
[20240903 10:45:03.477] [ui] [D] [5251427] - module  "com.bktus.gpgfrontend.module.gnupg_info_gathering" reader, read locale  "en_PL" , data size:  0
[20240903 10:45:03.477] [default] [W] [5251427] - QString::arg: Argument missing: :/i18n/ModuleVersionChecking.en_PL.qm loading, locale: en_PL, not found, com.bktus.gpgfrontend.module.version_checking
[20240903 10:45:03.477] [module] [W] [5251427] - ":/i18n/ModuleVersionChecking.en_PL.qm loading, locale: en_PL, not found"
[20240903 10:45:03.477] [ui] [D] [5251427] - module  "com.bktus.gpgfrontend.module.version_checking" reader, read locale  "en_PL" , data size:  0
[20240903 10:45:03.479] [ui] [D] [5251427] - main window start to initialize...
[20240903 10:45:03.542] [qt.qpa.fonts] [W] [5251427] - Populating font family aliases took 59 ms. Replace uses of missing font family "Courier" with one that exists to avoid this cost. 
[20240903 10:45:03.578] [core] [D] [5251427] - cannot find channel object, channel: 0
[20240903 10:45:03.579] [core] [D] [5251427] - start to load all cache from file system
[20240903 10:45:03.579] [core] [W] [5251427] - data object not found from disk, key:  "__cache_manage_data_register_key_list"
zsh: segmentation fault  /Applications/GpgFrontend.app/Contents/MacOS/GpgFrontend -l debug
```

Version
```shell
 $> /Applications/GpgFrontend.app/Contents/MacOS/GpgFrontend --version
GpgFrontend v2.1.4
Copyright (©) 2021 Saturneric <eric@bktus.com>
This is free software; see the source for copying conditions.

Build DateTime: 5 August 2024 17:04:43 UTC
Build Version: 2.1.4_darwin-23.5.0_arm64_release
Source Code Version: main_795a81bd1edad59ef51e73157b5ad68cfa30e328
```

MacOS 14.6.1 M2

**Comment by saturneric** [comment]
Please refer to: https://github.com/saturneric/GpgFrontend/issues/173

**Comment by nicraMarcin** [comment]
I don't know if it is directly related. It is similar but closed which means it is fixed but I still have this issue

**Comment by iPyGuy** [comment]
> I don't know if it is directly related. It is similar but closed which means it is fixed but I still have this issue

That’s because he applied the fix on August 16 but the latest version of the app was released on August 5, so it doesn’t contain the fix yet.

Wonder when a new version will be compiled and released?

**Comment by saturneric** [comment]
> > I don't know if it is directly related. It is similar but closed which means it is fixed but I still have this issue
> 
> That’s because he applied the fix on August 16 but the latest version of the app was released on August 5, so it doesn’t contain the fix yet.
> 
> Wonder when a new version will be compiled and released?

Sorry, I've been very busy these past two months, but I have some time now, so I'll probably release a new version in early October.

**Comment by saturneric** [comment]
Fixed in v2.1.5.
